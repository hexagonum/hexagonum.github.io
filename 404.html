<!DOCTYPE html><html lang="en"><head><title>HEXAGONUM</title><meta charSet="UTF-8"/><meta name="description" content="HEXAGONUM"/><meta name="viewport" content="width=device-width,initial-scale=1"/><link rel="icon" href="/favicon.ico"/><meta name="next-head-count" content="5"/><link rel="manifest" href="/manifest.json"/><link rel="apple-touch-icon" href="/icon.png"/> <meta name="theme-color" content="#000000"/><link rel="preload" href="/_next/static/css/82e206c0f6131cea.css" as="style"/><link rel="stylesheet" href="/_next/static/css/82e206c0f6131cea.css" data-n-g=""/><noscript data-n-css=""></noscript><script defer="" nomodule="" src="/_next/static/chunks/polyfills-c67a75d1b6f99dc8.js"></script><script src="/_next/static/chunks/webpack-8fa1640cc84ba8fe.js" defer=""></script><script src="/_next/static/chunks/framework-2c79e2a64abdb08b.js" defer=""></script><script src="/_next/static/chunks/main-b3b52fe1e13e8446.js" defer=""></script><script src="/_next/static/chunks/pages/_app-3cf1010c976be75d.js" defer=""></script><script src="/_next/static/chunks/pages/404-4f46368e2e47c1b7.js" defer=""></script><script src="/_next/static/tg4Aso3CgMTRTlUPoxTUb/_buildManifest.js" defer=""></script><script src="/_next/static/tg4Aso3CgMTRTlUPoxTUb/_ssgManifest.js" defer=""></script></head><body><div id="__next"><div class="bg-gray-900 text-gray-100"><div class="flex justify-center items-center h-screen"><div class="text-center"><h1 class="text-9xl font-bold">404</h1><p class="text-3xl uppercase">Page Not Found</p></div></div></div></div><script id="__NEXT_DATA__" type="application/json">{"props":{"pageProps":{}},"page":"/404","query":{},"buildId":"tg4Aso3CgMTRTlUPoxTUb","nextExport":true,"autoExport":true,"isFallback":false,"scriptLoader":[]}</script></body></html>