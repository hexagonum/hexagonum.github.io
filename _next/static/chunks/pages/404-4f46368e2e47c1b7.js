(self.webpackChunk_N_E=self.webpackChunk_N_E||[]).push([[197],{6141:function(e,n,t){(window.__NEXT_P=window.__NEXT_P||[]).push(["/404",function(){return t(7728)}])},7728:function(e,n,t){"use strict";t.r(n),t.d(n,{NotFoundPage:function(){return c}});var s=t(5893);let c=()=>(0,s.jsx)("div",{className:"bg-gray-900 text-gray-100",children:(0,s.jsx)("div",{className:"flex justify-center items-center h-screen",children:(0,s.jsxs)("div",{className:"text-center",children:[(0,s.jsx)("h1",{className:"text-9xl font-bold",children:"404"}),(0,s.jsx)("p",{className:"text-3xl uppercase",children:"Page Not Found"})]})})});n.default=c}},function(e){e.O(0,[774,888,179],function(){return e(e.s=6141)}),_N_E=e.O()}]);