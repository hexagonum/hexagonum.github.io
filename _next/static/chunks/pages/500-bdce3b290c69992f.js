(self.webpackChunk_N_E=self.webpackChunk_N_E||[]).push([[573],{698:function(e,n,r){(window.__NEXT_P=window.__NEXT_P||[]).push(["/500",function(){return r(2771)}])},2771:function(e,n,r){"use strict";r.r(n),r.d(n,{InternalServerErrorPage:function(){return s}});var t=r(5893);let s=()=>(0,t.jsx)("div",{className:"bg-gray-900 text-gray-100",children:(0,t.jsx)("div",{className:"flex justify-center items-center h-screen",children:(0,t.jsxs)("div",{className:"text-center",children:[(0,t.jsx)("h1",{className:"text-9xl font-bold",children:"500"}),(0,t.jsx)("p",{className:"text-3xl uppercase",children:"Internal Server Error"})]})})});n.default=s}},function(e){e.O(0,[774,888,179],function(){return e(e.s=698)}),_N_E=e.O()}]);