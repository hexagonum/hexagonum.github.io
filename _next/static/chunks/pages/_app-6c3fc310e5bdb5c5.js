(self.webpackChunk_N_E=self.webpackChunk_N_E||[]).push([[888],{6840:function(n,t,e){(window.__NEXT_P=window.__NEXT_P||[]).push(["/_app",function(){return e(9212)}])},9212:function(n,t,e){"use strict";e.r(t);var r=e(5893);e(8804);var u=e(9008),i=e.n(u);let c=n=>{let{Component:t,pageProps:e}=n;return(0,r.jsxs)(r.Fragment,{children:[(0,r.jsx)(i(),{children:(0,r.jsx)("title",{children:"HEXAGONUM"})}),(0,r.jsx)(t,{...e})]})};t.default=c},8804:function(){},9008:function(n,t,e){n.exports=e(2636)}},function(n){var t=function(t){return n(n.s=t)};n.O(0,[774,179],function(){return t(6840),t(6885)}),_N_E=n.O()}]);