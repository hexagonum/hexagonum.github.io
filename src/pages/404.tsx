import { NextPage } from 'next';

export const NotFoundPage: NextPage = () => {
  return (
    <div className="bg-gray-900 text-gray-100">
      <div className="flex justify-center items-center h-screen">
        <div className="text-center">
          <h1 className="text-9xl font-bold">404</h1>
          <p className="text-3xl uppercase">Page Not Found</p>
        </div>
      </div>
    </div>
  );
};

export default NotFoundPage;
