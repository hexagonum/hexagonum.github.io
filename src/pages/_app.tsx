import '@hexagonum/styles/globals.scss';
import type { AppProps } from 'next/app';
import Head from 'next/head';

const App: React.FC<AppProps> = ({ Component, pageProps }: AppProps) => {
  return (
    <>
      <Head>
        <title>HEXAGONUM</title>
        <meta charSet="UTF-8" key="charset" />
        <meta name="description" content="HEXAGONUM" />
        <meta
          name="viewport"
          content="width=device-width,initial-scale=1"
          key="viewport"
        />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <Component {...pageProps} />
    </>
  );
};

export default App;
