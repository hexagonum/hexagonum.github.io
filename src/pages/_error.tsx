import { NextPage } from 'next';

export const ErrorPage: NextPage = () => {
  return (
    <div className="bg-gray-900 text-gray-100">
      <div className="flex justify-center items-center h-screen">
        <div className="text-center">
          <h1 className="text-9xl font-bold">500</h1>
          <p className="text-3xl uppercase">Error</p>
        </div>
      </div>
    </div>
  );
};

export default ErrorPage;
