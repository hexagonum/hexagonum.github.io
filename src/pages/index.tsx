import Hexagon from '@hexagonum/components/Hexagon';
import useWindowSize from '@hexagonum/hooks/use-window-size';
import { NextPage } from 'next';
import Link from 'next/link';
import { useRouter } from 'next/router';
import { useEffect, useState } from 'react';

const themes = ['theme-0'];

type Project = { emoji: string; name: string; url: string };

const projects: Project[] = [
  {
    emoji: '☕',
    name: 'cafesang',
    url: 'https://hexagonum.github.io/cafesang/',
  },
  { emoji: '🎨', name: 'colors', url: 'https://hexagonum.github.io/colors/' },
  {
    emoji: '🧠',
    name: 'psychology',
    url: 'https://hexagonum.github.io/hofstede/',
  },
  {
    emoji: '🗺️',
    name: 'maps.svg',
    url: 'https://hexagonum.github.io/maps.svg/',
  },
  {
    emoji: '⚽',
    name: 'liverpool',
    url: 'https://hexagonum.github.io/liverpool/',
  },
  { emoji: '💻', name: 'forex', url: 'https://hexagonum.github.io/techstack/' },
  { emoji: '📈', name: 'trends', url: 'https://hexagonum.github.io/trends/' },
  { emoji: '☁️', name: 'weather', url: 'https://hexagonum.github.io/weather/' },
  { emoji: '🐝', name: 'bumblebee', url: '#' },
];

export const HomePage: NextPage = () => {
  const router = useRouter();
  const { visible } = router.query;

  const { width, height } = useWindowSize();

  const [theme, setTheme] = useState(themes[0]);
  const [{ maxRows, maxColumns }, setMax] = useState({
    maxColumns: Math.floor(width / 128) + 10,
    maxRows: Math.floor(height / 112) * 2 + 10,
  });

  useEffect(() => {
    const maxColumns = Math.floor(width / 128) + 5;
    const maxRows = Math.floor(height / 112) * 2 + 5;

    setMax({ maxColumns, maxRows });
  }, [width, height]);

  const columns = [...Array(maxColumns + 1).keys()].map((i: number) => i);
  const rows = [...Array(maxRows + 1).keys()].map(() => columns);

  return (
    <div className={theme}>
      <main className="w-screen h-screen overflow-hidden">
        <div className="relative">
          <div className="absolute top-0 right-0 left-0 bottom-0 mx-auto">
            <div className="relative">
              {rows.map((columns: number[], row: number) => {
                const filterProjects = projects.filter(
                  (_, index) => index % 2 === row % 2
                );
                return columns.map((column: number) => {
                  const projectIndex: number =
                    (column + row) % filterProjects.length;
                  const project = filterProjects[projectIndex];
                  return (
                    <button
                      type="button"
                      id={`hexagon-${row}-${column}`}
                      key={`hexagon-row-${column}`}
                      data-row={row}
                      data-column={column}
                      onClick={() => {
                        const themeIndex = themes.indexOf(theme);
                        const nextIndex =
                          themeIndex + 1 === themes.length ? 0 : themeIndex + 1;
                        const nextTheme = themes[nextIndex];
                        console.log(nextTheme);
                        setTheme(nextTheme);
                      }}
                      onMouseLeave={() => {
                        ['.level-1', '.level-2', '.level-3'].forEach(
                          (className) => {
                            const oldBoxes =
                              document.querySelectorAll(className);
                            oldBoxes.forEach((box) => {
                              box.classList.remove(className.replace('.', ''));
                            });
                          }
                        );
                      }}
                      onMouseOver={(event: any) => {
                        const row = parseInt(event.target.dataset.row, 10);
                        const column = parseInt(
                          event.target.dataset.column,
                          10
                        );
                        // Level 1
                        const leftLevel1 = row % 2 === 0 ? 0 : -1;
                        const rightLevel1 = row % 2 === 0 ? 1 : 0;
                        const level1Boxes = [
                          { row: row - 2, column: column },
                          { row: row - 1, column: column + leftLevel1 },
                          { row: row - 1, column: column + rightLevel1 },
                          { row: row + 1, column: column + leftLevel1 },
                          { row: row + 1, column: column + rightLevel1 },
                          { row: row + 2, column: column },
                        ].map(({ row, column }) => `hexagon-${row}-${column}`);
                        level1Boxes.forEach((id) => {
                          document.getElementById(id)?.classList.add('level-1');
                        });
                        // Level 2
                        const rightLevel2 = row % 2 ? 0 : 1;
                        const leftLevel2 = row % 2 ? -1 : 0;
                        const level2Boxes = [
                          { row: row - 4, column: column },
                          { row: row - 3, column: column + leftLevel2 },
                          { row: row - 3, column: column + rightLevel2 },
                          { row: row - 2, column: column + 1 },
                          { row: row - 2, column: column + -1 },
                          { row: row, column: column + -1 },
                          { row: row, column: column + 1 },
                          { row: row + 2, column: column + 1 },
                          { row: row + 2, column: column + -1 },
                          { row: row + 3, column: column + leftLevel2 },
                          { row: row + 3, column: column + rightLevel2 },
                          { row: row + 4, column: column },
                        ].map(({ row, column }) => `hexagon-${row}-${column}`);
                        level2Boxes.forEach((id) => {
                          document.getElementById(id)?.classList.add('level-2');
                        });
                        // Level 3
                        const level3Boxes = [
                          { row: row - 6, column: column },
                          { row: row - 5, column: column + leftLevel2 },
                          { row: row - 5, column: column + rightLevel2 },
                          { row: row - 4, column: column - 1 },
                          { row: row - 4, column: column + 1 },
                          { row: row - 3, column: column + leftLevel2 - 1 },
                          { row: row - 3, column: column + rightLevel2 + 1 },
                          { row: row - 1, column: column + leftLevel2 - 1 },
                          { row: row - 1, column: column + rightLevel2 + 1 },
                          { row: row + 1, column: column + leftLevel2 - 1 },
                          { row: row + 1, column: column + rightLevel2 + 1 },
                          { row: row + 3, column: column + leftLevel2 - 1 },
                          { row: row + 3, column: column + rightLevel2 + 1 },
                          { row: row + 4, column: column - 1 },
                          { row: row + 4, column: column + 1 },
                          { row: row + 5, column: column + leftLevel2 },
                          { row: row + 5, column: column + rightLevel2 },
                          { row: row + 6, column: column },
                        ].map(({ row, column }) => `hexagon-${row}-${column}`);
                        level3Boxes.forEach((id) => {
                          document.getElementById(id)?.classList.add('level-3');
                        });
                      }}
                      className="relative"
                    >
                      <div
                        className={`${
                          visible === 'true' ? 'flex' : 'project'
                        } absolute w-full h-full items-center justify-center text-6xl`}
                      >
                        <Link
                          id={project.name}
                          href={project.url}
                          target="_blank"
                        >
                          {project.emoji}
                        </Link>
                      </div>
                      <Hexagon />
                    </button>
                  );
                });
              })}
            </div>
          </div>
        </div>
      </main>
    </div>
  );
};

export default HomePage;
