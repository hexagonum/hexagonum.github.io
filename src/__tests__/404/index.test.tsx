import { NotFoundPage } from '@hexagonum/pages/404';
import { render } from '@testing-library/react';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({ query: { visible: 'true' } }),
}));

describe('NotFoundPage', () => {
  it('to match snapshot', () => {
    const { container } = render(<NotFoundPage />);
    expect(container).toMatchSnapshot();
  });
});
