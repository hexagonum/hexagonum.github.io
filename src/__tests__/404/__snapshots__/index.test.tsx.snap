// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage to match snapshot 1`] = `
<div>
  <div
    class="bg-gray-900 text-gray-100"
  >
    <div
      class="flex justify-center items-center h-screen"
    >
      <div
        class="text-center"
      >
        <h1
          class="text-9xl"
        >
          404
        </h1>
        <p
          class="text-3xl uppercase"
        >
          Page Not Found
        </p>
      </div>
    </div>
  </div>
</div>
`;
